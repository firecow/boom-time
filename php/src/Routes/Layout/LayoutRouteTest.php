<?
declare(strict_types=1);

namespace App\Routes\Layout;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\PHTML;
use App\Testing\RegressionTest;

class LayoutRouteTest extends RegressionTest
{
    /**
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $ctx = new ContextMock($superGlobals);
        $data = array(
            "title" => "SimplePHP",
            "subPageHTML" => "Empty sub page html",
            "headerHTML" => "Empty header html"
        );
        return PHTML::create("src/Routes/Layout/LayoutRoute.phtml", $data, $ctx);
    }
}
