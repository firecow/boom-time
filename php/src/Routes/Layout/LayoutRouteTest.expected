<!DOCTYPE html>
<html lang="en">
<head>

    <meta name="viewport" content="initial-scale=1">
    <title>SimplePHP</title>
    <link rel="shortcut icon" href="/favicon.png" type="image/x-icon"/>
    <link href="https://use.fontawesome.com/releases/v5.0.13/css/all.css" rel="stylesheet">
    <link rel="stylesheet" type="text/css" href="/style.css">

    <script type="text/javascript" src="/build/build.js"></script>
</head>
<body>

<div id="header" class="padding">
    Empty header html
</div>

<div id="subPage" class="padding">
    Empty sub page html
</div>

<div id="modal" class="modal centerMid fullSize hidden"></div>

<div id="loadingModal" class="hidden centerMid fullSize">
    <i class="fas fa-spinner fa-pulse bottomCenter"></i>
</div>

<div id="errorModal" class="modal centerMid fullSize hidden">

    <div id="errorModelContent" class="modal-content centerMid padding text-center">
        <h3>Der skete en fejl</h3>
        <div id="errorMessage"></div>
        <button onclick="return simplePHP.clickErrorModal.run(event);">
            <i class="fas fa-sync"></i>
            <c>Reload</c>
        </button>
    </div>

</div>

</body>
</html>